### PR TITLE
fix(github): harden CI repair identity, recovery and packaged workers

### PR DESCRIPTION
**Pull request by davincios**
Persisted CI repair runs must remain bound to the same GitHub account across username changes, and supervised workers must launch from both Python installs and packaged executables. This follow-up to #6229 also recovers interrupted registration and corrects retained report evidence.

#### Describe the changes you have made in this PR -

- Persist the stable numeric GitHub account ID and require it for report reads, active-run reuse and background execution. A renamed account retains access; a later holder of its old username does not. Legacy records without a numeric ID remain available locally but cannot authorize a remote account.
- Launch the worker through a hidden CLI command and share frozen-aware child command construction with scheduler service setup. Packaged executables receive CLI arguments rather than Python `-m` flags.
- Retire expired reservations only after the previous supervisor releases execution, allowing legacy interrupted runs to be replaced without adopting their reports or extending their deadlines.
- Recover a queued run interrupted before trigger registration, retaining its ID and original deadline. Previously registered stopped/removed schedules remain cancelled. Record registration without overwriting a worker that already started.
- Share failure classification with the CI verifier and require explicit SUCCESS for passing evidence links.
- Keep exception details in local logs; use fixed public errors for setup, worker and report-read failures. Preserve an actionable fixed message for an incompatible demo repository.

### Demo/Screenshot for feature changes and bug fixes -

```text
test_reports_require_the_recorded_github_account PASSED
test_interrupted_registration_recovers_original_run PASSED
test_registration_publish_preserves_an_already_started_worker PASSED
test_evidence_links_require_the_reported_outcome PASSED
test_setup_exception_details_stay_out_of_persisted_reports PASSED
test_worker_exception_details_stay_in_local_logs PASSED
test_frozen_child_command_is_parseable_and_obeys_the_saved_deadline PASSED
```

Validation: `make lint`, `make format-check`, `make typecheck`, and 151 focused repair lifecycle, verification, discovery, CLI startup/inventory, scheduler, packaging, constants and API-border tests. The original four review regressions and the account rename/reassignment regressions failed before their fixes. Tests use isolated stores, real scheduler/process lifecycle checks, and API-shaped fixtures; no live GitHub demo run is claimed.

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

Implemented and self-reviewed by Codex; this does not claim human review. The fixes stay in the GitHub repair domain and use existing credential injection, stable GitHub account IDs, check classification and locked persistence. The CLI worker imports through the GitHub public API and shares process command construction with scheduler setup. Public reports contain fixed messages while local logs keep diagnostic detail. Regression tests cover cross-account reads, interrupted registration, concurrent worker startup, evidence classification and exception sanitization.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the related PR
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

### CODE REVIEW

**Comment by davincios** [comment]
@greptile review

File: integrations/github/tools/ci_repair_loop/tool.py [hunk 1]
```diff
@@ -148,22 +153,32 @@ def schedule_ci_repair_loop(
...
     until = time.monotonic() + min(60, max(0, wait_seconds))
     try:
         store = RepairStore()
+        token = configured_token(github_token)
+        user = object_response(GitHubRestClient(token).request("GET", "user"))
+        actor = str(user.get("login") or "").casefold()
         while True:
             run = store.get(task_id)
+            if not actor or run.actor.casefold() != actor:
+                return {"ok": False, "error": "This repair belongs to a different GitHub account."}
```
[the remark this answers is not in the document]
Comment (davincios) [reply integrations/github/tools/ci_repair_loop/tool.py:169]: Fixed in ba89a6f2e. Runs now persist GitHub’s numeric account ID, and scheduling reuse, report access, and the worker all compare that stable identity. Legacy records without an ID fail authorization rather than adopting the current holder of a login. Regression tests reproduced rejection after a rename and unauthorized reuse/read/write with a reassigned login; all now pass. Local lint, format, typecheck, packaging/CLI and focused lifecycle tests are green.

**Comment by davincios** [comment]
@greptile review

File: integrations/github/tools/ci_repair_loop/storage.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ def reserve(self, candidate: RepairRun) -> tuple[RepairRun, bool]:
             runs = self._read()
             for run in runs.values():
                 if run.identity[1:] == candidate.identity[1:] and not run.terminal:
-                    if run.actor.casefold() != candidate.actor.casefold():
+                    if not run.actor_id or run.actor_id != candidate.actor_id:
                         raise ValueError(
                             "Another GitHub account already has an active repair for this target."
                         )
```
[the remark this answers is not in the document]
Comment (davincios) [reply integrations/github/tools/ci_repair_loop/storage.py]: Fixed in 67c8220eb. Reservation now retires expired runs only while holding the execution lock, preserving their identity and deadline as terminal history before creating a fresh run. A supervisor that is still stopping blocks replacement briefly; an unexpired legacy run remains unauthorized until its original deadline. The regression loads an actual pre-ID JSON record, proves an active supervisor prevents replacement, then proves safe replacement and stable subsequent reuse after shutdown. The repair lifecycle suite (24 tests), lint, formatting and typecheck pass.

**Comment by davincios** [comment]
@greptile review
